src/server/procedures/myForestV2/leaderboard.ts

```diff
@@ -0,0 +1,109 @@
+import { z } from 'zod';
+import { procedure } from '../../trpc';
+import { Leaderboard } from '../../../features/common/types/myForestv2';
+import prisma from '../../../../prisma/client';
+import { TRPCError } from '@trpc/server';
+import { Prisma } from '@prisma/client';
+import { fetchProfile } from '../../utils/fetchProfile';
+import { fetchGroupTreecounterData } from '../../utils/fetchGroupTreecounterData';
+
+async function fetchMostRecentGifts(profileIds: number[]) {
+  return await prisma.$queryRaw<
+    {
+      quantity: number;
+      giverName: string;
+    }[]
+  >`
+		SELECT
+			round((value)/100, 2) as quantity,
+			COALESCE(NULLIF(metadata->>'$.giver.name', ''), 'anonymous') as giverName
+		FROM gift
+		WHERE 
+			recipient_id IN (${Prisma.join(profileIds)}) 
+			AND deleted_at IS NULL 
+			AND value <> 0
+		ORDER BY payment_date DESC
+		LIMIT 10;
+	`;
+}
+
+async function fetchTopGifters(profileIds: number[]) {
+  return await prisma.$queryRaw<
+    {
+      totalQuantity: number;
+      giverName: string;
+    }[]
+  >`
+		SELECT
+			sum(round((value)/100, 2)) as totalQuantity,
+			metadata->>'$.giver.name' as giverName
+		FROM gift
+		WHERE 
+			recipient_id IN (${Prisma.join(profileIds)}) 
+			AND deleted_at IS NULL 
+			AND value <> 0 
+			AND metadata->>'$.giver.name' IS NOT NULL
+			AND metadata->>'$.giver.name' <> ''
+		GROUP BY giverName
+		ORDER BY totalQuantity DESC
```

Comment:
Extract the gift-related SQL queries into a separate service or utility module to keep the procedure clean

Reply:
I prefer to leave it as is because:

- They aren't used anywhere else, and are already within isolated functions. 
- Moving them elsewhere will make the code less readable, as I'd need to move between files to see the functionality. 

I've already kept some functions that could be reused in a utils folder.